vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/features/metadata/source/exif/ExifMetadataExtractor.kt

```diff
@@ -0,0 +1,63 @@
+package org.vitrivr.engine.core.features.metadata.source.exif
+
+import com.drew.imaging.ImageMetadataReader
+import com.drew.metadata.Directory
+import com.drew.metadata.Metadata
+import org.vitrivr.engine.core.features.AbstractExtractor
+import org.vitrivr.engine.core.model.content.element.ContentElement
+import org.vitrivr.engine.core.model.descriptor.struct.metadata.source.MapStructDescriptor
+import org.vitrivr.engine.core.model.metamodel.Schema
+import org.vitrivr.engine.core.model.retrievable.Retrievable
+import org.vitrivr.engine.core.model.retrievable.attributes.SourceAttribute
+import org.vitrivr.engine.core.operators.Operator
+import org.vitrivr.engine.core.source.MediaType
+import org.vitrivr.engine.core.source.file.FileSource
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+
+class ExifMetadataExtractor(
+    input: Operator<Retrievable>,
+    field: Schema.Field<ContentElement<*>, MapStructDescriptor>,
+    persisting: Boolean = true
+) : AbstractExtractor<ContentElement<*>, MapStructDescriptor>(input, field, persisting, bufferSize = 1) {
+    override fun matches(retrievable: Retrievable): Boolean =
+        retrievable.filteredAttribute(SourceAttribute::class.java)?.source is FileSource /* && retrievable.type is MediaType.IMAGE */
```

Comment:
Why is the image type requirement commented? As far as I understand, EXIF is exclusively a standard for images.

Reply:
I can comment on this since this line goes back to my contribution: I intended to have an image filter at this point for the reason you mention, yet in the way it is written there it does not work and I was unsure how to write it in a manner idiomatic for vitrivr-engine.

Reply:
This review is based on what I can access on github, hence I could not check the vitrivr-engine idomatic way of things, maybe @ppanopticon or @lucaro could enlighten us?

Reply:
According to https://github.com/drewnoakes/metadata-extractor, the library extracts far more than just EXIF, so it would also be applicable for video, not only for images.
`retrievable.type is MediaType.IMAGE` makes no sense since `retrievable.type` is a `String`, which cannot possibly inherit from `MediaType`. You would want something like `(retrievable.filteredAttribute(SourceAttribute::class.java)?.source as FileSource).type == MediaType.IMAGE`, but then again, you'd also want to check for videos.

Reply:
EXIF is indeed image only, the used library does however also support various other metadata standards, including ones for video containers.